x-pack/plugins/ml/public/cases/single_metric_viewer_attachment.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiDescriptionList } from '@elastic/eui';
+import type { PersistableStateAttachmentViewProps } from '@kbn/cases-plugin/public/client/attachment_framework/types';
+import { ReactEmbeddableRenderer } from '@kbn/embeddable-plugin/public';
+import type { FieldFormatsStart } from '@kbn/field-formats-plugin/public';
+import { FIELD_FORMAT_IDS } from '@kbn/field-formats-plugin/common';
+import { FormattedMessage } from '@kbn/i18n-react';
+import deepEqual from 'fast-deep-equal';
+import { memoize } from 'lodash';
+import React from 'react';
+import { CASE_ATTACHMENT_TYPE_ID_SINGLE_METRIC_VIEWER } from '../../common/constants/cases';
+import type {
+  SingleMetricViewerEmbeddableApi,
+  SingleMetricViewerEmbeddableState,
+} from '../embeddables/types';
+
+export const initComponent = memoize((fieldFormats: FieldFormatsStart) => {
+  return React.memo(
+    (props: PersistableStateAttachmentViewProps) => {
+      const { persistableStateAttachmentState, caseData } = props;
+
+      const inputProps =
+        persistableStateAttachmentState as unknown as SingleMetricViewerEmbeddableState;
+
+      const dataFormatter = fieldFormats.deserialize({
+        id: FIELD_FORMAT_IDS.DATE,
+      });
+
+      const listItems = [
+        {
+          title: (
+            <FormattedMessage
+              id="xpack.ml.cases.singleMetricViewer.description.jobIdsLabel"
+              defaultMessage="Job IDs"
+            />
+          ),
+          description: inputProps.jobIds.join(', '),
+        },
+        {
+          title: (
+            <FormattedMessage
+              id="xpack.ml.cases.singleMetricViewer.description.timeRangeLabel"
+              defaultMessage="Time range"
+            />
+          ),
+          description: `${dataFormatter.convert(
+            inputProps.timeRange!.from
+          )} - ${dataFormatter.convert(inputProps.timeRange!.to)}`,
+        },
+      ];
+
+      if (typeof inputProps.query?.query === 'string' && inputProps.query?.query !== '') {
+        listItems.push({
+          title: (
+            <FormattedMessage
+              id="xpack.ml.cases.singleMetricViewer.description.queryLabel"
+              defaultMessage="Query"
+            />
+          ),
+          description: inputProps.query?.query,
+        });
+      }
+
+      return (
+        <>
+          <EuiDescriptionList compressed type={'inline'} listItems={listItems} />
+          <ReactEmbeddableRenderer<
```

Comment:
Can you make it work without `ReactEmbeddableRenderer`? We should avoid using it in general, anomaly swim lane in an exception and I should change it soon. 
You can check how the change point detection attachment looks like https://github.com/elastic/kibana/blob/main/x-pack/plugins/aiops/public/cases/change_point_charts_attachment.tsx

Reply:
Thanks for pointing this out - didn't realize we were trying to move away from it.
Removed use of `ReactEmbeddableRenderer`, created `single_metric_viewer.tsx` as a separate component that can be passed to the embeddable factory and the case attachment.
Also moved the `mlServices` loading into a separate function so it can be used for what the `single_metric_viewer` needs.
 4c2d66866dcc36466a48b92a9b02fb7e77e146c4